### PR TITLE
#274 Task/resolve password issue

### PR DESCRIPTION
**Pull request by SzanieckiFranciszek**
#### Checklist template for LEADERS

&#x2611; | &#x2610;

&#x2611; New code is covered with tests and coverage requirements are met.
&#x2610; Build is not failing.
&#x2610; All tests are passing.
&#x2610; All conflicts are resolved and the branch is rebased on top of the up-to-date parent branch.
&#x2611; All issue requirements are implemented.
&#x2610; At least two reviewers are assigned.
&#x2611; The Pull request is linked to the related issue.
&#x2611; Correct branch name. Convention [https://github.com/bycza-zagroda/track-expenses-app-documentation/blob/develop/githubwork/GITHUB_WORK.md]
&#x2611; Correct unit test name - suffix 'Test' in the end.
&#x2611; Correct integration test name - suffix 'IT' in the end.
&#x2611; Positive and negative test paths have been implemented.

### CODE REVIEW

**Review by dawciobiel — DISMISSED** [review]
LGTM

**Review by Iwona007 — CHANGES_REQUESTED** [review]
Please look at UserServiceImpl class at lines: you don't inject @Value from config and  you create a new Instant for pattern. 
1. Please delete line 90 and 96 and as parameters/arguments use this value from application lines: 42, 43
2. Line 38 -BCryptPasswordEncoder already exist, please delete line 114. - (This is after merge develop into feature branch)
3. Next time when you resolve conflict, please pay attention for a differences between both branches

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/AuthLoginDTO.java
Comment (mirekgab) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/AuthLoginDTO.java:1]: Check if this works as we expected.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/AuthRegisterDTO.java
Comment (mirekgab) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/AuthRegisterDTO.java:1]: Check if this works as we expected.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java [hunk 1]
```diff
@@ -56,11 +56,11 @@ public class User implements Serializable {
 
     @NotBlank
     @Size(min = 6, max = 120)
-    @Pattern(regexp = "^[\\w-.]+@([\\w-]+.)+[\\w-]{2,4}$")
+    @Value("${regex.pattern.email}")
```
Comment (mirekgab) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java:59]: Check if this works as we expected.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java [hunk 2]
```diff
@@ -56,11 +56,11 @@ public class User implements Serializable {
...
     private String email;
 
     @NotBlank
-    @Size(min = 8, max = 100)
+    @Value("${regex.pattern.password}")
```
Comment (mirekgab) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java:63]: Is regex needed here?

File: src/main/resources/db/changelog/data/users.csv
Comment (mirekgab) [inline src/main/resources/db/changelog/data/users.csv:1]: Add comments with a plain text password so we can log in and test these users.
Examples:
```
id,user_name,email,password,user_status
#password is asdf
1,ScroogeMcDuck,scroogeMcDuck@gmail.com,e3c2e42987be081a9eccf744eb90726eeb7255691184308cafe2968646ca6073,VERIFIED

#password is
2,Heuy,heuy@gmail.com,a32065a955ccea2b3e18d3c6486954eca77b64a904660fb256c5047fbd49dd5e,VERIFIED
```

**Review by mirekgab — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/data/users.csv
Comment (SzanieckiFranciszek) [reply src/main/resources/db/changelog/data/users.csv:1]: Done.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java (hunk 2 above)
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java:63]: It is not necessary by the fact that the same regex is in DTOs but I go on the assumption that it is better to add. If I have to delete please tell me.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java (hunk 1 above)
Comment (mirekgab) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java:59]: From the spring documentation for the Value annotation:

Note that actual processing of the @Value annotation is performed by a [BeanPostProcessor](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanPostProcessor.html) which in turn means that you cannot use @Value within [BeanPostProcessor](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanPostProcessor.html) or [BeanFactoryPostProcessor](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanFactoryPostProcessor.html) types. Please consult the javadoc for the [AutowiredAnnotationBeanPostProcessor](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/annotation/AutowiredAnnotationBeanPostProcessor.html) class (which, by default, checks for the presence of this annotation).

This means that the Value annotation applies to the bean, not the entity. Additionally, the Value annotation has a different purpose than the Pattern annotation.

File: src/main/resources/db/changelog/data/users.csv
Comment (mirekgab) [reply src/main/resources/db/changelog/data/users.csv:1]: This hashes are not valid.
You can use https://www.javainuse.com/onlineBcrypt to encrypt passwords.

Comment (SzanieckiFranciszek) [reply src/main/resources/db/changelog/data/users.csv:1]: Corrected.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java (hunk 2 above)
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/model/User.java:63]: Yes, you have right.  I removed validation with pattern in User class.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/AuthRegisterDTO.java
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/AuthRegisterDTO.java:1]: checked, works well .

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/AuthLoginDTO.java
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/AuthLoginDTO.java:1]: checked, works well .

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/UserServiceImpl.java [hunk 3]
```diff
@@ -135,4 +136,18 @@ private Cookie createRefreshTokenCookie(User user) {
         return cookie;
     }
 
+    private void validatePasswordLength(String password) {
+
+        if (password.length() < 8) {
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/UserServiceImpl.java:141]: Why still not using contains for "magic value" - `8`?
as well for `100` value below.

Please change it.

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Request changes

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/UserServiceImpl.java (hunk 3 above)
Comment (SzanieckiFranciszek) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/service/UserServiceImpl.java:141]: Updated.

**Review by mirekgab — DISMISSED** [review]

**Review by dawciobiel — DISMISSED** [review]
LGTM

**Review by Iwona007 — APPROVED** [review]

**Review by mirekgab — APPROVED** [review]

**Pull request merged**